scheduler/test/cook/test/simulator.clj

```diff
@@ -0,0 +1,599 @@
+         riemann-port# (:riemann-port ~scheduler-config)
+         gpu-enabled?# (or (:gpus-enabled? ~scheduler-config) false)
+         rebalancer-config# (merge default-rebalancer-config (:rebalancer-config ~scheduler-config))
+         fenzo-config# (merge default-fenzo-config (:fenzo-config ~scheduler-config))
+         trigger-chans# (or (:trigger-chans ~scheduler-config)
+                            (c/make-trigger-chans rebalancer-config# task-constraints#))]
+     (try
+       (let [scheduler# (c/start-mesos-scheduler ~make-mesos-driver-fn get-mesos-utilization#
+                                                 curator-framework# ~conn
+                                                 mesos-mult# zk-prefix#
+                                                 offer-incubate-time-ms# mea-culpa-failure-limit#
+                                                 task-constraints# riemann-host# riemann-port#
+                                                 pending-jobs-atom# offer-cache#
+                                                 gpu-enabled?# rebalancer-config# fenzo-config#
+                                                 trigger-chans#)]
+         ~@body)
+       (finally
+         (.close curator-framework#)
+         (.stop zookeeper-server#)
+         (doseq [[trigger-service# trigger-chan#] trigger-chans#]
+           (log/info "Shutting down" trigger-service#)
+           (async/close! trigger-chan#))))))
+
+(defn generate-task-trace-map
+  [task]
+  (let [job (:job/_instance task)
+        resources (util/job-ent->resources job)]
+    {:job_id  (str (:job/uuid job))
+     :instance_id  (:instance/task-id task)
+     :submit_time_ms  (.getTime (:job/submit-time job))
+     :mesos_start_time_ms  (if (:instance/mesos-start-time task)
+                             (.getTime (:instance/mesos-start-time task))
+                             -1)
+     :start_time_ms  (.getTime (:instance/start-time task))
+     :end_time_ms  (.getTime (or (:instance/end-time task) (tc/to-date (t/now))))
+     :status  (:instance/status task)
+     :hostname  (:instance/hostname task)
+     :slave_id  (:instance/slave-id task)
+     :reason  (or (when (= (:instance/status task) :instance.status/failed)
+                    (:reason/string (:instance/reason task)))
+                  "")
+     :user  (:job/user job)
+     :mem  (or (:mem resources) -1)
+     :cpus  (or (:cpus resources) -1)
+     :job_name  (or (:job/name job) "")
+     :requested_run_time  (->> (:job/label job)
+                               (filter #(= (:label/key %) "JOB-RUNTIME"))
+                               first
+                               :label/value)
+     :requested_status  (->> (:job/label job)
+                             (filter #(= (:label/key %) "JOB-STATUS"))
+                             first
+                             :label/value)}))
+
+(defn dump-jobs-to-csv
+  "Given a mesos db, dump a csv with a row per task
+
+   Columns:
```

Comment:
I prefer order unless there is a good reason not to. Looks like there is a good reason here.